### PR TITLE
Create, Edit, Clone API calls

### PR DESCRIPTION
**Pull request by ankittrehan2000**

### CODE REVIEW

File: web/src/components/QuestionForm.jsx [hunk 1]
```diff
@@ -58,32 +58,43 @@ const useStyles = makeStyles(theme => ({
...
     answer: '',
     instructions: [],
+    gameId
   });
   useEffect(() => {
-    if (originalQuestion) setQuestion(originalQuestion)
+    if (originalQuestion) {
+      if (originalQuestion.instructions !== null && originalQuestion.instructions !== [] && typeof originalQuestion.instructions === 'string') {
+        originalQuestion.instructions = JSON.parse(originalQuestion.instructions);
+        originalQuestion.instructions = JSON.parse(originalQuestion.instructions);
```
[the remark this answers is not in the document]
Comment (ankittrehan2000) [reply web/src/components/QuestionForm.jsx:77]: This was intentional, for some reason when the instructions come back from AWS, it needed to be parsed twice

File: web/src/components/QuestionForm.jsx [hunk 2]
```diff
@@ -105,18 +114,18 @@ function QuestionForm({ loading, saveQuestion, question: originalQuestion, quest
...
+        {question.imageUrl && <img className={classes.image} src={question.imageUrl} alt="Preview" />}
       </div>
 
       <Divider className={classes.divider} />
 
       <TextField className={classes.input} id="answer" value={question.answer} onChange={onChangeMaker('answer')} label="Answer" variant="filled" required />
       <h3>Solution Steps</h3>
       <List>
-        {question.instructions.map((step, index) => (
+        {typeof question.instructions != "string" && question?.instructions?.map((step, index) => (
```
[the remark this answers is not in the document]
Comment (ankittrehan2000) [reply web/src/components/QuestionForm.jsx:128]: I did try this but there wasn't a way to do `typeof string`

[the remark this answers is not in the document]
Comment (maniramezan) [reply web/src/components/QuestionForm.jsx:128]: JS being JS I guess 🙄

File: web/src/components/QuestionForm.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (maniramezan) [reply web/src/components/QuestionForm.jsx:77]: hmm, this sounds weird, what's the API response?

[the remark this answers is not in the document]
Comment (ankittrehan2000) [reply web/src/components/QuestionForm.jsx:77]: "[\"Test step\"]"

**Comment by ankittrehan2000** [comment]
I think there is still some things that Lucah is working on but wanted to open it (and possibly get this reviewed and merged) before the rest of the web team gets working on the teacher app. This is because they will need the latest API calls in order to work through the UI updates

**Review by maniramezan — APPROVED** [review]
LGTM

**Pull request merged**